### PR TITLE
CORE-10346 Updating kafka topic name from RPC -> REST

### PR DESCRIPTION
**Pull request by ben-millar**
This PR is part of the broader RPC -> REST renaming initiative described in [CORE-7040](https://r3-cev.atlassian.net/browse/CORE-7040), and satisfies [CORE-8034](https://r3-cev.atlassian.net/browse/CORE-8034) (rename `rpc-worker` -> `rest-worker`), and [CORE-10346](https://r3-cev.atlassian.net/browse/CORE-10346) (rename kafka topic from `rpc` -> `rest` worker).

This PR goes hand-in-hand with the runtime-os changes in this PR: https://github.com/corda/corda-runtime-os/pull/3110

# PR Checklist:

- [ ] Have you run the unit, integration and smoke tests as described [here](https://docs.corda.net/head/testing.html)?
- [ ] If you added public APIs, did you write the JavaDocs?
- [ ] If the changes are of interest to application developers, have you added them to the [changelog](https://docs.corda.net/head/changelog.html) (`/docs/source/changelog.rst`), and potentially the [release notes](https://docs.corda.net/head/release-notes.html) (`/docs/source/release-notes.rst`)?
- [ ] If you are contributing for the first time, please read the [contributor agreement](https://docs.corda.net/head/contributing.html) now and add a comment to this pull request stating that your PR is in accordance with the [Developer's Certificate of Origin](https://docs.corda.net/head/contributing.html#merging-the-changes-back-into-corda).

Thanks for your code, it's appreciated! :)

[CORE-7040]: https://r3-cev.atlassian.net/browse/CORE-7040?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CORE-8034]: https://r3-cev.atlassian.net/browse/CORE-8034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[CORE-10346]: https://r3-cev.atlassian.net/browse/CORE-10346?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: data/topic-schema/src/main/kotlin/net/corda/schema/Schemas.kt [hunk 1]
```diff
@@ -162,16 +162,16 @@ class Schemas {
...
+    class Rest {
         companion object {
-            const val RPC_PERM_MGMT_REQ_TOPIC = "rpc.permissions.management"
-            val RPC_PERM_MGMT_RESP_TOPIC = getRPCResponseTopic(RPC_PERM_MGMT_REQ_TOPIC)
-            const val RPC_PERM_USER_TOPIC = "rpc.permissions.user"
-            const val RPC_PERM_GROUP_TOPIC = "rpc.permissions.group"
-            const val RPC_PERM_ROLE_TOPIC = "rpc.permissions.role"
-            const val RPC_PERM_ENTITY_TOPIC = "rpc.permissions.permission"
+            const val REST_PERM_MGMT_REQ_TOPIC = "rest.permissions.management"
+            val REST_PERM_MGMT_RESP_TOPIC = getRPCResponseTopic(REST_PERM_MGMT_REQ_TOPIC)
```
Comment (mnesbit) [inline data/topic-schema/src/main/kotlin/net/corda/schema/Schemas.kt:170]: Definitely check that the DB worker part of this works and is using these same constants. Do some grepping globally for these strings

File: gradle.properties [hunk 2]
```diff
@@ -7,7 +7,7 @@ cordaProductVersion = 5.0.0
 # NOTE: update this each time this module contains a breaking change
 ## NOTE: currently this is a top level revision, so all API versions will line up, but this could be moved to
 ##   a per module property in which case module versions can change independently.
-cordaApiRevision = 646
+cordaApiRevision = 140995
```
Comment (mnesbit) [inline gradle.properties]: Looks very wrong!

**Review by mnesbit — DISMISSED** [review]

File: data/topic-schema/src/main/kotlin/net/corda/schema/Schemas.kt (hunk 1 above)
Comment (ben-millar) [reply data/topic-schema/src/main/kotlin/net/corda/schema/Schemas.kt:170]: Good question! This change has been made everywhere; the previous `RPC_PERM_*` exists only in dated build outputs.

File: gradle.properties [hunk 3]
```diff
@@ -7,7 +7,7 @@ cordaProductVersion = 5.0.0
 # NOTE: update this each time this module contains a breaking change
 ## NOTE: currently this is a top level revision, so all API versions will line up, but this could be moved to
 ##   a per module property in which case module versions can change independently.
-cordaApiRevision = 649
+cordaApiRevision = 651
```
Comment (ben-millar) [inline gradle.properties:10]: Version number has now been set correctly as discussed.

**Review by mnesbit — APPROVED** [review]
OK

**Pull request merged**